app/models/package_manager/base.rb

```diff
@@ -178,12 +178,18 @@ def self.add_version(db_project, version_hash)
       end
     end
 
-    def self.deprecate_versions(db_project, version_hash)
+    def self.deprecate_versions(db_project, version_hashes)
       case db_project.platform.downcase
       when "rubygems" # yanked gems will be omitted from project JSON versions
         db_project
           .versions
-          .where.not(number: version_hash.pluck(:number))
+          .where.not(number: version_hashes.pluck(:number))
+          .update_all(status: "Removed")
+      when "npm"
+        not_deprecated_version_hashes = version_hashes.select { |v| v[:is_deprecated] == false }
+        db_project
+          .versions
+          .where.not(number: not_deprecated_version_hashes.pluck(:number))
```

Comment:
are we marking versions that are still downloadable, but are deprecated, as removed? Is there a backstory on that?

Reply:
That was the intent behind this code. Do we not want to do that? I could remove that functionality if we don't actually want it, but I had assumed that since there doesn't appear to be an explicit "yanked" denotion for NPM that I'm aware of then maybe deprecated was the mostly equivalent option.

Reply:
Hm, maybe we don't actually want deprecated versions, I'm not sure. I had assumed deprecated on NPM was similar to the way yanked was on Rubygems. But looking at a [yanked version on Rubygems](https://rubygems.org/gems/flowbyte-yanked/versions/1.0.1), even though it's visible it's not downloadable, whereas looking at a [deprecated version on NPM](https://www.npmjs.com/package/react/v/0.7.1), it is still downloadable.

Do we want me to just remove the deprecation checking behavior here, and just instead only look at the diff between the versions we have in the DB and the versions we receive in quantity from the NPM API response?

Reply:
yes I think they are different. Deprecated means it's available but someone said not to use it for reasons, yanked/removed means it's not available for download.